indexing-handlers/src/main/java/no/unit/nva/indexing/model/IndexRequest.java

```diff
@@ -23,14 +25,30 @@ public IndexRequest(String name, String jsonMappings) {
         };
         this.mappings = attempt(() -> JsonUtils.dtoObjectMapper.readValue(jsonMappings, typeReference))
                 .orElseThrow();
+        this.settings = Collections.emptyMap();
     }
 
+    public IndexRequest(String name, String jsonMappings, String jsonSettings) {
+        this.name = name;
+        var typeReference = new TypeReference<Map<String, Object>>() {
+        };
+        this.mappings = attempt(() -> JsonUtils.dtoObjectMapper.readValue(jsonMappings, typeReference))
+            .orElseThrow();
+        this.settings = attempt(() -> JsonUtils.dtoObjectMapper.readValue(jsonSettings, typeReference))
+            .orElseThrow();
```

Comment:
Had to do a double-take here as I imagined I was seeing double :D Can you extract the mapping + type reference to its own method `mapToJavaType(String jsonString)`